### PR TITLE
Speedup build: introduced SharedSqlTestDatabase and DisposableSqlTestDatabase junit extensions

### PR DESCRIPTION
**Pull request by a-sayyed**
Extracted shared database logic from PR https://github.com/eclipse/hawkbit/pull/1126 (goal state).

This is an intermediate state, where the tests
- reuse the generated databases - one generated db per jvm process (`surefire.forkCount`)
- use a single tenant for all the tests
- Have the `CleanupTestExecutionListener` run after each method

Only two tests require an own database: `SystemManagementTest` and `MultiTenancyEntityTest` since they modify the default tenant

### CODE REVIEW

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/SystemManagementTest.java [hunk 1]
```diff
@@ -20,15 +23,23 @@
...
+import org.springframework.test.context.TestExecutionListeners;
 
 import io.qameta.allure.Description;
 import io.qameta.allure.Feature;
 import io.qameta.allure.Story;
+import net.jcip.annotations.NotThreadSafe;
 
 @Feature("Component Tests - Repository")
 @Story("System Management")
+@NotThreadSafe // test methods should not run in parallel, since they use the same tenant
```
Comment (a-sayyed) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/SystemManagementTest.java:41]: remove, not needed

File: hawkbit-repository/hawkbit-repository-test/src/main/java/org/eclipse/hawkbit/repository/test/util/DatasourceContext.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    public DatasourceContext(final Class<?> clazz) {
+        database = getFromEnvironmentOrDefault(clazz, SPRING_DATABASE_KEY, "H2");
+        datasourceUrl = getFromEnvironmentOrDefault(clazz, SPRING_DATASOURCE_URL_KEY, "jdbc:h2:mem:test;MODE=MySQL;");
+        username = getFromEnvironmentOrDefault(clazz, SPRING_DATABASE_USERNAME_KEY, "sa");
+        password = getFromEnvironmentOrDefault(clazz, SPRING_DATABASE_PASSWORD_KEY, "");
+    }
+
+    /**
+     * if system property is set, it will be used, otherwise, property will be loaded from {@link HAWKBIT_JPA_DEFAULTS_LOCATION}
```
Comment (a-sayyed) [inline hawkbit-repository/hawkbit-repository-test/src/main/java/org/eclipse/hawkbit/repository/test/util/DatasourceContext.java:63]: todo: remove hardcoded location, and read `PropertySource` annotation instead

**Review by herdt-michael — APPROVED** [review]
Looks good, thank you @a-sayyed and @floruschbaschan !

**Comment by a-sayyed** [comment]
I can not believe my eyes! 🍾

**Pull request merged**